src/herder/HerderImpl.cpp

```diff
@@ -455,6 +457,20 @@ HerderImpl::valueExternalized(uint64 slotIndex, StellarValue const& value,
                          gap, fw.write(slotInfo));
         }
 
+        auto slotInfo = getSCP().getJsonQuorumInfo(getSCP().getLocalNodeID(),
+                                                   false, true, slotIndex);
+        std::set<std::string> missing;
+        for (const auto& node : slotInfo["missing"])
+        {
+            missing.insert(node.asString());
+        }
+        std::set<std::string> prevMissing = std::move(mMissingNodes);
+        mMissingNodes.clear();
+        std::set_intersection(
+            missing.begin(), missing.end(), prevMissing.begin(),
+            prevMissing.end(),
+            std::inserter(mMissingNodes, mMissingNodes.begin()));
```

Comment:
A couple things:
1. Can you please add a comment at the start of this chunk explaining what you're doing (e.g. pruning `mMissingNodes` by removing any nodes heard from in the new slot).
2. Does this pruning only run when the node is in sync? That's fine if so, but we might want to change the way we report dead nodes if the node is not in sync (such as maybe not reporting anything when out of sync).